test/src/test/java/org/corfudb/runtime/LogReplicationLogicalGroupClientShimTest.java

```diff
@@ -0,0 +1,330 @@
+import org.corfudb.runtime.LogReplication.ClientRegistrationId;
+import org.corfudb.runtime.LogReplication.ClientRegistrationInfo;
+import org.corfudb.runtime.LogReplication.DestinationInfoVal;
+import org.corfudb.runtime.collections.CorfuStoreShim;
+import org.corfudb.runtime.collections.Table;
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.view.AbstractViewTest;
+import org.corfudb.test.SampleSchema.ManagedResources;
+import org.corfudb.test.SampleSchema.Uuid;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+public class LogReplicationLogicalGroupClientShimTest extends AbstractViewTest {
+
+    private CorfuRuntime getTestRuntime() {
+        return getDefaultRuntime();
+    }
+    private static String namespace;
+    private static String registrationTableName;
+    private static String metadataTableName;
+    private static String clientName;
+    private static CorfuRuntime runtime;
+    private static CorfuStoreShim store;
+
+    @Before
+    public void loadProperties() throws Exception {
+        namespace = "CorfuSystem";
+        registrationTableName = "LogReplicationRegistrationTable";
+        metadataTableName = "LogReplicationModelMetadataTable";
+        clientName = "client";
+
+        runtime = getTestRuntime();
+        store = new CorfuStoreShim(runtime);
+    }
+
+    /**
+     * Test registering replication client
+     *
+     */
+    @Test
+    public void testRegisterReplicationClient() throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+        final Table<ClientRegistrationId, ClientRegistrationInfo, ManagedResources> table = store.openTable(
+                namespace,
+                registrationTableName,
+                ClientRegistrationId.class,
+                ClientRegistrationInfo.class,
+                ManagedResources.class,
+                TableOptions.fromProtoSchema(Uuid.class));
+
+        // Test registering client with null/empty client name
+        Assert.assertThrows(IllegalArgumentException.class, () ->  new LogReplicationLogicalGroupClient(runtime, null));
+        Assert.assertThrows(IllegalArgumentException.class, () ->  new LogReplicationLogicalGroupClient(runtime, ""));
+
+        // Test registering a new client
+        new LogReplicationLogicalGroupClient(runtime, clientName);
+        final int expectedNumberRegisteredClients = 1;
+        Assert.assertEquals(expectedNumberRegisteredClients, table.count());
+
+        // Test registering a duplicate client
+        new LogReplicationLogicalGroupClient(runtime, clientName);
+        final int expectedNumberRegisteredClients1 = 1;
```

Comment:
nit : can continue to use expectedNumberRegisteredClients here

Reply:
I think I have a few instances where some variables are redundant, figured it would be more readable if each assert had its own expected variable

Reply:
can you rename it to something more explicit then?  expectedNumberRegisterClients1 is not very intuitive...

Reply:
That was the plan initially but variable names became too verbose for some scenarios and hard to read with naming inconsistencies. So I opted for consistent variable names local to each scenario with a descriptor comment right above